src/item.h

```diff
@@ -427,6 +427,14 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
      * Puts the given item into this one, no checks are performed.
      */
     void put_in( item payload );
+
+    /** Stores a newly constructed item at the end of this item's contents */
+    template<typename ... Args>
+    item& emplace_in( Args&&... args ) {
```

Comment:
Personal opinion: the function name `emplace_back` is better, it matches the name of the function in the standard library and as such is a bit more intuitive for those knowing that. For everyone else, neither `emplace_back` nor `emplace_in` makes a difference.